File: MAP_Lab7/MAP_Lab7/Repository/IPrgStateRepo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MAP_Lab7.Statements;

namespace MAP_Lab7.Repository
{
    public interface IPrgStateRepo
    {
        void Add(PrgState st);
        PrgState GetCurrentProgram();
        void logPrgStateExec();
    }
}

File: Toy Interpreter/src/Repository/IPrgStateRepo.java
/**************************************************************************************************
 *       Copyright(c):                                                                            *
 *                         Program made by @Moldovan Alexandru-Vasile.                            *
 *                                                                                                *
 **************************************************************************************************/

package Repository;
import Model.Statement.PrgState;
import javafx.collections.ObservableList;

import java.util.List;

public interface IPrgStateRepo
{
    void addPrgState(PrgState st);
    PrgState getCurrentProgram();
    void logPrgStateExec(PrgState prgState);
    List<PrgState> getPrgList();
    void setPrgList(List<PrgState> list);
    ObservableList<String> getIDs();
}
